src/symbol/mod.rs

```diff
@@ -6,67 +6,25 @@
 // copied, modified, or distributed except according to those terms.
 
 use std::fmt;
-use std::result;
+
+use scroll::{ctx::TryFromCtx, Endian, Pread, LE};
 
 use crate::common::*;
 use crate::msf::*;
 use crate::FallibleIterator;
 
+mod annotations;
 mod constants;
+
 use self::constants::*;
 
-/// PDB symbol tables contain names, locations, and metadata about functions, global/static data,
-/// constants, data types, and more.
-///
-/// The `SymbolTable` holds a `SourceView` referencing the symbol table inside the PDB file. All the
-/// data structures returned by a `SymbolTable` refer to that buffer.
-///
-/// # Example
-///
-/// ```
-/// # use pdb::FallibleIterator;
-/// #
-/// # fn test() -> pdb::Result<usize> {
-/// let file = std::fs::File::open("fixtures/self/foo.pdb")?;
-/// let mut pdb = pdb::PDB::open(file)?;
-///
-/// let symbol_table = pdb.global_symbols()?;
-/// let address_map = pdb.address_map()?;
-///
-/// # let mut count: usize = 0;
-/// let mut symbols = symbol_table.iter();
-/// while let Some(symbol) = symbols.next()? {
-///     match symbol.parse() {
-///         Ok(pdb::SymbolData::PublicSymbol(data)) if data.function => {
-///             // we found the location of a function!
-///             let rva = data.offset.to_rva(&address_map).unwrap_or_default();
-///             println!("{} is {}", rva, symbol.name()?);
-///             # count += 1;
-///         }
-///         _ => {}
-///     }
-/// }
-///
-/// # Ok(count)
-/// # }
-/// # assert!(test().expect("test") > 2000);
-/// ```
-#[derive(Debug)]
-pub struct SymbolTable<'s> {
-    stream: Stream<'s>,
-}
+pub use self::annotations::*;
 
-impl<'s> SymbolTable<'s> {
-    /// Parses a symbol table from raw stream data.
-    pub(crate) fn parse(stream: Stream<'s>) -> Result<Self> {
-        Ok(SymbolTable { stream })
-    }
+/// The raw type discriminator for `Symbols`.
+pub type SymbolKind = u16;
 
-    /// Returns an iterator that can traverse the symbol table in sequential order.
-    pub fn iter(&self) -> SymbolIter<'_> {
-        SymbolIter::new(self.stream.parse_buffer())
-    }
-}
+/// A register referred to by its number.
+pub type Register = u16;
```

Comment:
Hmm… what's the user going to do with this? If it warrants support functions to decode, would `struct Register(pub u16)` be better?

Reply:
https://github.com/willglynn/pdb/pull/47#discussion_r320677524